interfaces/molecule/packmol.py

```diff
@@ -748,6 +764,230 @@ def packmol_in_void(
     return ret
 
 
+def _run_uff_md(
+    ucs: "ChemicalSystem",
+    nsteps: int = 1000,
+    vectors=None,
+    fixed_atoms: Optional[Sequence[int]] = None,
+    keepjob: bool = False,
+) -> "ChemicalSystem":
+    """
+    Runs UFF MD with SHAKE all bonds, keeps ``fixed_atoms`` (0-based atom indices) fixed,
+    if ``vectors`` is not None will transform into those vectors
+
+    Returns: The final system from the MD simulation.
+
+    Raises: PackmolError if something goes worng.
+    """
+    from scm.plams.core.functions import delete_job
+    from scm.plams import config
+
+    thermostatted_region = "PACKMOL_thermostatted"
+    md_ucs = ucs.copy()
+    md_ucs.set_atoms_in_region(
+        [x for x in range(len(md_ucs)) if fixed_atoms is None or x not in fixed_atoms], thermostatted_region
+    )
+
+    s = Settings()
+    s.input.ForceField.Type = "UFF"
+    s.input.ams.Task = "MolecularDynamics"
+    if fixed_atoms:
+        s.input.ams.Constraints.AtomList = " ".join(str(x + 1) for x in fixed_atoms)
+    s.input.ams.MolecularDynamics.NSteps = nsteps
+    s.input.ams.MolecularDynamics.TimeStep = 0.5
+    s.input.ams.MolecularDynamics.Shake.All = "bonds * *"
+    s.input.ams.MolecularDynamics.InitialVelocities.Type = "Zero"
+    # s.input.ams.MolecularDynamics.InitialVelocities.Temperature = 10
+    s.input.ams.MolecularDynamics.Thermostat.Temperature = 5
+    s.input.ams.MolecularDynamics.Thermostat.Region = thermostatted_region
+    s.input.ams.MolecularDynamics.Thermostat.Tau = 2
+    s.input.ams.MolecularDynamics.Thermostat.Type = "Berendsen"
+
+    if vectors is not None:
+        x = vectors
+        target_lattice_str = f"""
+            {x[0][0]} {x[0][1]} {x[0][2]}
+            {x[1][0]} {x[1][1]} {x[1][2]}
+            {x[2][0]} {x[2][1]} {x[2][2]}
+        """
+
+        s.input.ams.MolecularDynamics.Deformation.StartStep = 1
+        s.input.ams.MolecularDynamics.Deformation.StopStep = (nsteps * 3) // 4
+        s.input.ams.MolecularDynamics.Deformation.TargetLattice._1 = target_lattice_str
+
+    previous_config = config.copy()
+    config.job.pickle = False
+    config.log.stdout = 0
+    # TODO: fix this so that it doesn't leave plams_workdir on disk if it is created
+    job = AMSJob(settings=s, molecule=md_ucs, name="shakemd")
+    job.run()
+    job.results.wait()
+    config.job.pickle = previous_config.job.pickle
+    config.log.stdout = previous_config.log.stdout
+    if not job.ok():
+        raise PackMolError(
+            f"Try a lower density or a less skewed cell! Original file in {job.path} . "
+            + str(job.results.get_errormsg())
+        )
+    my_packed = job.results.get_main_system()
+    if not keepjob:
+        delete_job(job)
+
+    my_packed.remove_region("PACKMOL_thermostatted")
```

Comment:
I would suggest something like:

```
    previous_config = config.copy()
    try:
        config.job.pickle = False
        config.log.stdout = 0

        with tempfile.TemporaryDirectory() as tmp_dir:
            job_manager = JobManager(config.jobmanager, path=tmp_dir)
            job = AMSJob(settings=s, molecule=md_ucs, name="shakemd")
            job.run(job_manager=job_manager)

            if not job.ok():
                raise PackMolError(
                    f"Try a lower density or a less skewed cell! Original file in {job.path} . "
                    + str(job.results.get_errormsg())
                )
            print(f"***{tmp_dir}")
            my_packed = job.results.get_main_system()
            my_packed.remove_region("PACKMOL_thermostatted")

    finally:
        config.job.pickle = previous_config.job.pickle
        config.log.stdout = previous_config.log.stdout
```
which uses a scratch job manager that runs this job in a tmp dir that is then cleaned up.

Note that this makes the `keepjob` option obsolete. Note also that the `keepjob` is set to `True` as noted below, so that the job is currently being kept in the created workdir anyway.

If you want to have the option of still keeping the workdir, instead of a random tmp dir I would use a dir with a random name in the current directory, which you can choose to clean up (or not).

Reply:
done (except remove underscore in job_manager)